src/obi_auth/util.py

```diff
@@ -0,0 +1,42 @@
+"""Utilities."""
+
+import base64
+import hashlib
+import platform
+import socket
+from pathlib import Path
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+
+
+def get_machine_salt():
+    """Get machine specific salt."""
+    hostname = socket.gethostname()
+    system = platform.uname().system
+    node = platform.uname().node
+    raw = f"{hostname}-{system}-{node}"
```

Comment:
On my machine this gives just `work.local-Darwin-work.local`.
If there isn't a simple cross-platform way to get a unique machine id, maybe we can add the username to make it a bit more specific to the user?

Reply:
I changed the salt to include as much information about the system and the user as possible keeping it cross-platform.